From c1a605c8c7af872b01603cfbe80cff4f51894ea3 Mon Sep 17 00:00:00 2001
From: Gu Zitao <guzitao@wxiat.com>
Date: Thu, 14 Jul 2022 10:00:10 +0800
Subject: [PATCH] sw64: add MIGHT_HAVE_PC_SERIO option to control selection of
 i8042
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: c1a605c8c7af872b01603cfbe80cff4f51894ea3
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

Since using i8042 drivers on sw64 would cause kernel crash, we add
MIGHT_HAVE_PC_SERIO option to control selection of i8042.

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/Kconfig | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/sw_64/Kconfig b/arch/sw_64/Kconfig
index f03be9ce50cc..f9a270396f5a 100644
--- a/arch/sw_64/Kconfig
+++ b/arch/sw_64/Kconfig
@@ -14,7 +14,6 @@ config SW64
 	select GENERIC_IRQ_SHOW
 	select ARCH_WANT_IPC_PARSE_VERSION
 	select ARCH_HAVE_NMI_SAFE_CMPXCHG
-	select ARCH_MIGHT_HAVE_PC_SERIO
 	select ARCH_NO_PREEMPT
 	select ARCH_USE_CMPXCHG_LOCKREF
 	select GENERIC_SMP_IDLE_THREAD
@@ -241,6 +240,11 @@ config PLATFORM_XUELANG
 
 endchoice
 
+config MIGHT_HAVE_PC_SERIO
+	bool "Use PC serio device i8042"
+	select ARCH_MIGHT_HAVE_PC_SERIO
+	default n
+
 endmenu
 
 config LOCK_MEMB
-- 
2.33.0

